framework/File.cpp

```diff
@@ -73,40 +73,40 @@ int FS_WriteFloatString( char *buf, const char *fmt, va_list argPtr ) {
 							index += sprintf( buf+index, "%s", tmp.c_str() );
 						}
 						else {
-							index += sprintf( buf+index, format.c_str(), f );
+							index += sprintf( buf+index, "%s", format.c_str(), f );
```

Comment:
This seems incorrect, and after taking some time away from keyboard, it seems clearer why: the format-string provided to a bunch of these statements only contains a one replacement pattern (i.e. `%s` - replace with one string, whereas `%s %s` would be replace with one string, then a space, then a second string)

Using a sub-call to `va` here is a possible workaround if the format-string first argument to `sprintf` is to be kept.

Reply:
You probably shouldn't change this function.
As the name `format` suggests, it's the format string to be used here.
As far as I can tell, this function basically wraps sprintf(), and replicates the behavior of the original (by calling the original with `fmt`'s format specifiers), except that it prints floating point numbers with a higher accuracy.
Real ugly and kinda fragile, don't touch ;)

Here it might make sense to use the pragma to shut up the warnings

Reply:
Note: I'm planning on following [this pattern](https://github.com/blendogames/quadrilateralcowboy/blob/e9ad2007df77a7c0679c00f0528837d43ce299af/game/Game_local.cpp#L80-L82) (check GCC version, add a pragma within a .cpp file) instead of a header declaration to add the initial warning filtering.

If it ends up being applicable in a bunch more places, perhaps a global header definition would make sense, but starting with a small local definition seemed to make sense to me.